src/cache/ElementQueryTagDependency.php

```diff
@@ -18,20 +18,14 @@
  */
 class ElementQueryTagDependency extends TagDependency
 {
-    /**
-     * @var ElementQuery|null
-     */
-    public ?ElementQuery $elementQuery = null;
-
     /**
      * Constructor
      *
      * @param ElementQuery $elementQuery
      * @param array $config
      */
-    public function __construct(ElementQuery $elementQuery, array $config = [])
+    public function __construct(public ?ElementQuery $elementQuery, array $config = [])
```

Comment:
This changes the signature. `null` was previously not allowed to be passed to `$elementQuery`.

Reply:
The promotion property is  based on property type, so you choose: to have nullable or non-nullable property?

Reply:
We should not be making any type changes as part of this PR, either to properties or method arguments.

Reply:
sure, skipped property promotion for this file https://github.com/craftcms/cms/pull/16711/commits/2ce4511b49c8611ddbac10ea6180bdafedbbca9e 👍